packages/db/src/translate/granules.ts

```diff
@@ -171,8 +172,11 @@ const validateApiToPostgresGranuleObject = (apiGranule : ApiGranule) => {
   if (isNil(apiGranule.granuleId)) {
     throw new ValidationError('granuleId cannot be undefined on a granule, granules must have a collection and a granule ID');
   }
+  if (isNil(apiGranule.producerGranuleId)) {
```

Comment:
Nit: might be worth the time to refactor dynamo -> api in the code that calls this (might not too!)